src/AutorestSwift/Models/Complex/ConstantSchema.swift

```diff
@@ -54,7 +58,28 @@ class ConstantSchema: Schema {
         var container = encoder.container(keyedBy: CodingKeys.self)
 
         try container.encode(valueType, forKey: .valueType)
-        try container.encode(value, forKey: .value)
+
+        var valueContainer = container
+            .nestedContainer(keyedBy: ValueCodingKeys.self, forKey: .value)
+
+        switch valueType.type {
+        case .boolean:
+            try valueContainer.encode(Bool(value.value), forKey: .value)
+        case .integer:
+            try valueContainer.encode(Int(value.value), forKey: .value)
+        case .number:
+            if let numberSchema = valueType as? NumberSchema {
+                if numberSchema.precision >= 32 {
+                    try valueContainer.encode(Double(value.value), forKey: .value)
+                } else {
+                    try valueContainer.encode(value.value, forKey: .value)
+                }
+            } else {
+                try valueContainer.encode(Double(value.value), forKey: .value)
+            }
+        default:
+            try valueContainer.encode(value.value, forKey: .value)
+        }
```

Comment:
`This logic should live in the encode method for ConstantValue.swift. `. I can't put this logic in ConstantValue.swift since it does not have the `type` information there. The logic is in ConstantSchema , since it has both `type` and `vaule`

Reply:
Interesting. Then perhaps ConstantValue should be moved as an internal struct of ConstantSchema to ensure it can't be instantiated by itself.

Reply:
Move ConstantValue to inside Constant.